benchmarks/src/jmh/java/com/linecorp/armeria/thrift/StackTracingBenchmark.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.thrift;
+
+import java.lang.reflect.Constructor;
+import java.security.Permission;
+import java.util.function.Supplier;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+import com.linecorp.armeria.thrift.services.HelloService.hello_args;
+
+/**
+ * Compare performance of Thread.getStackTrace vs StackWalker(reflected) vs StackWalker(reflected) with Option
+ * find getCurrentThriftMessageClass.
+ *
+ * <p>20190928 Macbook Pro 2018 2.2 GHz Intel Core i7
+ * <pre>
+ * # Run complete. Total time: 00:25:10
+ *
+ * Benchmark                                          Mode  Cnt       Score       Error  Units
+ * StackTracingBenchmark.defaultStackTrace           thrpt   25   65969.836 ±  4714.734  ops/s
+ * StackTracingBenchmark.usingStackWalker            thrpt   25  131919.125 ±  7730.472  ops/s
+ * StackTracingBenchmark.usingStackWalkerWithOption  thrpt   25  325797.300 ± 33808.269  ops/s
+ *
+ * </pre>
+ */
+
+@State(Scope.Benchmark)
+public class StackTracingBenchmark extends hello_args {
+
+    private Supplier<Class<?>> defaultThriftMessageClassFinderfault;
+    private Supplier<Class<?>> stackWalkingThriftMessageClassFinder;
+    private Supplier<Class<?>> stackWalkingFinderWithRetainOption;
+
+    @Setup
+    public void makeFinder() throws Exception {
+        final Class<?> defaultThriftMessageClassFinderClass =
+                Class.forName("com.linecorp.armeria.common.thrift.text.DefaultThriftMessageClassFinder");
+        final Class<?> stackWalkingThriftMessageClassFinderClass =
+                Class.forName("com.linecorp.armeria.common.thrift.text.StackWalkingThriftMessageClassFinder");
```

Comment:
Perhaps we do not use reflection if we put this benchmark class in the same package?

Reply:
I didn't think that. thanks for your advice.